src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -421,7 +421,18 @@ def dispatch_to_local_service(
     service, method = _look_up_service_method(service_name, method_name)
     raw_arguments = service.deserialize_rpc_arguments(method_name, serial_arguments)
     result = method(**raw_arguments.__dict__)
-    return result.dict() if isinstance(result, RpcModel) else result
+
+    if isinstance(result, RpcModel):
+        return result.dict()
+    if isinstance(result, list):
```

Comment:
Minor concern 1: This commits us further to using `List` as the type annotation for all serializable collections. There seem to be some practical issues with passing `Collection` or `Sequence` to Pydantic, which I haven't fully investigated. But I don't feel great about creating an edge case here in case we use a `tuple` or `set` in the future.

Could we instead check if it's any iterable other than `str`?